test/Fund.test.js

```diff
@@ -0,0 +1,234 @@
+    await this.endaomentAdmin.setRole(2, accountant, { from: admin });
+    await this.endaomentAdmin.setRole(3, reviewer, { from: admin });
+    this.fund = await Fund.new(manager, this.endaomentAdmin.address, {
+      from: admin,
+    });
+  });
+
+  it("has defined fund contract address post-init", async function() {
+    assert.isDefined(this.fund.address);
+  });
+
+  it("allows ADMIN to construct Fund contract", async function() {
+    const fund = await Fund.new(manager, this.endaomentAdmin.address, {
+      from: admin,
+    });
+    assert.isDefined(fund.address);
+  });
+
+  it("allows FUND_FACTORY to construct Fund contract via ACCOUNTANT", async function() {
+    const fundFactory = await FundFactory.new(this.endaomentAdmin.address, {
+      from: admin,
+    });
+    await this.endaomentAdmin.setRole(4, fundFactory.address, { from: admin });
+    const fund = await fundFactory.createFund(
+      manager,
+      this.endaomentAdmin.address,
+      {
+        from: accountant,
+      }
+    );
+    assert.isDefined(fund.logs[0].args.newAddress);
+  });
+
+  it("denies invalid admin contract to construct fund contract", async function() {
+    await expectRevert.unspecified(
+      Fund.new(constants.ZERO_ADDRESS, this.endaomentAdmin.address, {
+        from: admin,
+      })
+    );
+  });
+
+  it("denies invalid non-ADMIN wallet to construct fund contract", async function() {
+    await expectRevert.unspecified(
+      Fund.new(manager, this.endaomentAdmin.address, { from: manager })
+    );
+  });
+
+  it("creates funds with correct manager", async function() {
+    const fund_contract = await Fund.at(this.fund.address);
+    const fund_manager = await fund_contract.manager();
+
+    assert.equal(fund_manager, manager);
+    assert.isDefined(this.fund.address);
+  });
+
+  it("allows ADMIN to change manager", async function() {
+    const fund_contract = await Fund.at(this.fund.address);
+    const fund_manager = await fund_contract.manager();
+
+    assert.equal(fund_manager, manager);
+
+    await fund_contract.changeManager(newManager, this.endaomentAdmin.address, {
+      from: admin,
+    });
+    const new_manager = await fund_contract.manager();
+
+    await assert.equal(new_manager, newManager);
+  });
+
+  it("allows REVIEWER to change manager, only if not paused", async function() {
+    const fund_contract = await Fund.at(this.fund.address);
+    const fund_manager = await fund_contract.manager();
+
+    assert.equal(fund_manager, manager);
+
+    await fund_contract.changeManager(newManager, this.endaomentAdmin.address, {
+      from: reviewer,
+    });
+    const new_manager = await fund_contract.manager();
+
+    await assert.equal(new_manager, newManager);
+
+    await this.endaomentAdmin.pause(3, { from: pauser });
+
+    await expectRevert.unspecified(
+      fund_contract.changeManager(newManager, this.endaomentAdmin.address, {
+        from: reviewer,
+      })
+    );
+  });
```

Comment:
split into two test:

1. allows REVIEWER to change manager
2. denies REVIEWER to change manager when paused